.github/workflows/nix.yml

```
@@ -0,0 +1,20 @@
+name: Nix derivation checks
+
+on:
+  push:
+    paths: ["mix.lock"]
```

We probably want to only do this on `main`. I think this would do it?

```suggestion
on:
  push:
    paths: ["mix.lock"]
    branches: ["main"]
```